core/commands/UserGetCommand.php

```diff
@@ -0,0 +1,167 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'authentication_api.php' );
+require_api( 'config_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that gets user information. If id is not specified, it will get the current user.
+ * To get other users' information, the user must have manage_user_threshold access level.
+ *
+ * Sample:
+ * {
+ *   "query": {
+ *      "id": 1234
+ *   },
+ *   "options": {
+ * 	    "include_in_user_element": true
+ *   }
+ * }
+ */
+class UserGetCommand extends Command {
+	/**
+	 * @var integer The id of the user to get.
+	 */
+	private $target_user_id;
+
+	/**
+	 * Fields to show if no `select` query parameter is specified.
+	 * 
+	 * @var array
+	 */
+	private $select;
+
+	# TODO: add support for enabled flag
+	# TODO: add supoprt for protected flag
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		$t_current_user_id = auth_get_current_user_id();
+		$t_user_id = $this->query( 'id', null );
+		if( !is_null( $t_user_id ) && $t_user_id <= 0 ) {
+			throw new ClientException(
+				"Invalid user id '$t_user_id'",
+				ERROR_INVALID_FIELD_VALUE,
+				array( 'id' ) );
+		}
+
+		$t_user_ref = $this->query( 'ref', null );
+
+		$this->target_user_id = 0;
+
+		if( is_null( $t_user_id ) ) {
+			if( is_null( $t_user_ref ) ) {
+				$this->target_user_id = $t_current_user_id;
+			} else {
+				$t_lookup_user_id = user_get_id_by_name( $t_user_ref );
+				if( $t_lookup_user_id !== false ) {
+					$this->target_user_id = $t_lookup_user_id;
+				} else {
+					$t_lookup_user_id = user_get_id_by_email( $t_user_ref );
+					if( $t_lookup_user_id !== false ) {
+						$this->target_user_id = $t_lookup_user_id;
+					} else {
+						$t_lookup_user_id = user_get_id_by_realname( $t_user_ref );
+						if( $t_lookup_user_id !== false ) {
+							$this->target_user_id = $t_lookup_user_id;
+						} else if( is_numeric( $t_user_ref ) ) {
+							$this->target_user_id = (int)$t_user_ref;
+						}
+					}
+				}
+			}
+		} else {
+			$this->target_user_id = (int)$t_user_id;
+		}
+
+		if( !is_null( $t_user_ref ) && $this->target_user_id == 0 ) {
+			throw new ClientException(
+				"User not found with id, name, realname or email equal to '$t_user_ref'",
+				ERROR_USER_BY_NAME_NOT_FOUND,
```

Comment:
We actually have distinct error codes for the various scenarios (ERROR_USER_BY_ID_NOT_FOUND, ERROR_USER_BY_EMAIL_NOT_FOUND, ERROR_USER_BY_REALNAME_NOT_FOUND), so the Exception should use them and not throw a potentially incorrect ERROR_USER_BY_NAME_NOT_FOUND.

Set a variable with the error code in the if blocks above.

Reply:
I have changed the API to be:

/users/me
/users/:user_id
/users/username/:username

And remove by email and real name.